app/client/lib/ACUserManager.ts

```diff
@@ -0,0 +1,225 @@
+        save(emailObs.get());
+        finish();
+        return true;
+      }
+    } catch (e) {
+        emailInput.setCustomValidity(e.message);
+        return false;
+    } finally {
+      emailInput.reportValidity();
+    }
+  };
+
+  const onMouseDown = (ev: MouseEvent) => {
+    ev.preventDefault(); // Don't let it affect focus, since we focus/blur manually.
+    if (!isOpen()) {
+      emailInput.focus();
+    }
+    openOrCommit();
+  };
+
+  const maybeShowAddNew = async (results: ACResults<ACUserItem>, text: string): Promise<ACResults<ACUserItem>> => {
+    const cleanText = normalizeText(text);
+    const items = results.items
+      .filter(item => item.cleanText.includes(cleanText))
+      .sort((a,b) => a.cleanText.localeCompare(b.cleanText));
+    results.items = items;
+    if (!results.items.length) {
+      const newObject = {
+        value: text,
+        cleanText,
+        name: "",
+        email: "",
+        isNew: true,
+        label: text,
+        id: 0,
+      };
+      results.items.push(newObject);
+    }
+    return results;
+  };
+
+  const renderSearchItem = (item: ACUserItem, highlightFunc: any): HTMLLIElement => (item?.isNew ? cssSelectItem(
+    cssMemberListItem(
+      cssUserImagePlus(
+        "+",
+        cssUserImage.cls("-large"),
+        cssUserImagePlus.cls('-invalid', (use) => !use(enableAdd),
+      )),
+      cssMemberText(
+        cssMemberPrimaryPlus("Invite new member"),
+        cssMemberSecondaryPlus(
+          dom.text(use => `We'll email an invite to ${use(emailObs)}`)
+        )
+      ),
+      testId("um-add-email")
+    )
+  ) : cssSelectItem(
+    cssMemberListItem(
+      cssMemberImage(createUserImage(item, "large")),
+      cssMemberText(
+        cssMemberPrimaryPlus(item.name, testId("um-member-name")),
+        cssMemberSecondaryPlus(buildHighlightedDom(item.label, highlightFunc, cssMatchText))
+      )
+    )
+  ));
+
+  const enableAdd: Computed<boolean> = computed((use) => Boolean(use(emailObs) && use(isInputValid)));
+
+  const acOptions: IAutocompleteOptions<ACUserItem> = {
+    menuCssClass: `${menuCssClass} test-acselect-dropdown`,
+    search: (term) => maybeShowAddNew(acIndex.search(term), term),
+    renderItem: renderSearchItem,
+    getItemText: (item) => item.value,
+    onClick: commitIfValid,
+  };
+
+  const result = cssEmailInputContainer(
+    dom.autoDispose(enableAdd),
+    cssMailIcon("Mail"),
+    (emailInput = cssEmailInput(
+      emailObs,
+      {onInput: true, isValid: isInputValid},
+      { type: "email", placeholder: "Enter email address" },
+      dom.on("input", acOpen),
+      dom.on("focus", (_ev, elem) => elem.select()),
+      dom.on("blur", commitOrRevert),
+      dom.onKeyDown({
+        Escape: revert,
+        Enter: openOrCommit,
+        ArrowDown: acOpen,
+        Tab: commitIfValid,
+      }),
+      cssEmailInputContainer.cls('-green', enableAdd),
+      ...args
+    )),
+    dom.on("mousedown", onMouseDown)
```

Comment:
```suggestion
      ...args
    )),
    cssEmailInputContainer.cls('-green', enableAdd),
```

I think the `cls` was in the wrong spot. With this change, the input container now turns green again when the email is valid.